### ISSUE TITLE
Upgrade failing

### ISSUE THREAD

**Issue by mattpackwood** [serverless/serverless#8493]
⚠️⚠️ REPLACE WITH DESCRIPTION

<details>
<summary><code>serverless.yml</code></summary>

```yaml
Not yet configured
```

</details>

<details>
<summary><b><code>⚠️⚠️ REPLACE WITH FULL COMMAND NAME</code> output</b></summary>

```
Matt-Packwoods-iMac:~ $ serverless update
 
  Serverless Error ---------------------------------------
 
  Serverless command "update" not found. Did you mean "create"? Run "serverless help" for a list of all available commands.
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information ---------------------------
     Operating System:          darwin
     Node Version:              12.18.1
     Framework Version:         2.9.0 (standalone)
     Plugin Version:            4.1.1
     SDK Version:               2.3.2
     Components Version:        3.2.7
```

</details>

<b>Installed version</b>

```
Framework Core: 2.9.0 (standalone)
Plugin: 4.1.1
SDK: 2.3.2
Components: 3.2.7```

**Comment by medikoo** [comment]
@mattpackwood command name is `serverless upgrade`

**Comment by mattpackwood** [comment]
@medikoo Hi, issue reoccurring after full reinstall...

**Comment by medikoo** [comment]
@mattpackwood have you seen my comment above?

**Comment by mattpackwood** [comment]
Sorry, yes, I typoed while I was writing, I am typing "serverless upgrade", and getting the error, 

 `Error: EISDIR: illegal operation on a directory, open '/var/folders/s7/typ53r597p33k6rpcwn9lg_40000gn/T'`

Matt

**Comment by medikoo** [comment]
This looks as if user with which you're executing the command has no access to temporary folders in which application may store temporary files.

Why do you think it's the case?

Still what can be considered  bug on our side, is that we do not report such errors meaningfully (we may also explore alternatives a storing temp files in user folder, but that feels dirty)

**Comment by mattpackwood** [comment]
Hmmm, that is interesting, I am running the command as an administrator but not su on MacOS, but that does not explain the issue in WSL, should I be trying to execute it as su "everywhere" or change the permissions on the relevant folder?

**Comment by medikoo** [comment]
@mattpackwood normallly shoud not need sudo for anything, and I believe for normal configurations on macOS there are no issue. I personally work on macOS, do not use sudo, and have no problems in accessing temporary folders.

There must be some specific issue in your setup that raises the trouble

**Comment by mattpackwood** [comment]
I looked at the folders and the permissions look OK I am going to continue to experiment with it on some different hardware/set ups.  I can uninstall and reinstall to get updates, so I will keep trying different things...

**Comment by budnieswski** [comment]
I'm having the same issue, and already check the folders permissions.

Folders:


```
~ serverless upgrade
Serverless: Downloading new version...

  Error --------------------------------------------------

  Error: EISDIR: illegal operation on a directory, open '/var/folders/tv/bs3lmyb159q9y4cs8lh4b_fr0000gn/T'

     For debugging logs, run again after setting the "SLS_DEBUG=*" environment variable.

  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com

  Your Environment Information ---------------------------
     Operating System:          darwin
     Node Version:              12.18.1
     Framework Version:         2.14.0 (standalone)
     Plugin Version:            4.1.2
     SDK Version:               2.3.2
     Components Version:        3.4.2
```

**Comment by budnieswski** [comment]
I'm having the same issue, and already check folders permissions.

<details>
<summary><code>Folders</code></summary>

```
/usr/local/bin
/usr/local/lib
~/.npm
```

</details>

<details>
<summary><code>Output</code></summary>

```
~ serverless upgrade
Serverless: Downloading new version...

  Error --------------------------------------------------

  Error: EISDIR: illegal operation on a directory, open '/var/folders/tv/bs3lmyb159q9y4cs8lh4b_fr0000gn/T'

     For debugging logs, run again after setting the "SLS_DEBUG=*" environment variable.

  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com

  Your Environment Information ---------------------------
     Operating System:          darwin
     Node Version:              12.18.1
     Framework Version:         2.14.0 (standalone)
     Plugin Version:            4.1.2
     SDK Version:               2.3.2
     Components Version:        3.4.2
```

</details>

**Comment by medikoo** [comment]
@budnieswski thanks for additional information.

I've looked more closer, and indeed there's a bug in temporary path handling, a regression introduced with https://github.com/serverless/serverless/pull/8428 (published with v2.9.0). Fix will be issued shortly
